packaging/linux/openra-server.in

```diff
@@ -1,4 +1,10 @@
 #!/bin/sh
-cd "{GAME_INSTALL_DIR}"
+cd "{GAME_INSTALL_DIR}" || exit
 
-mono {DEBUG} OpenRA.Server.dll Game.Mod={MODID} "$@"
+if test -x "OpenRA.Server"; then
+	./OpenRA.Server Game.Mod={MODID} "$@"
+elif command -v mono >/dev/null 2>&1 && [ "$(grep -c .NETCoreApp,Version= OpenRA.Server.dll)" = "0" ]; then
+	mono --debug OpenRA.Server.dll Game.Mod={MODID} "$@"
+else
+	dotnet OpenRA.Server.dll Game.Mod={MODID} "$@"
+fi
```

Comment:
Maybe install small scripts **OpenRA** and **OpenRA.Server** for the mono version. Then you can always start with `./OpenRA ...`.

I was thinking of adding something like

```sh
printf '#!/bin/sh\nexec mono --debug %s.dll "$@"\n' "$0" >"${DEST_PATH}/OpenRA.Server"
chmod +x "${DEST_PATH}/OpenRA.Server"
if [ "${COPY_GENERIC_LAUNCHER}" = "True" ]; then
    cp "${DEST_PATH}/OpenRA.Server" "${DEST_PATH}/OpenRA"
fi
````

to [`install_assemblies()`](https://github.com/OpenRA/OpenRA/blob/6a31b1f9f368109cc1916ba6513f4c0eb30cf8b6/packaging/functions.sh#L38-L79).

Reply:
I like this idea because the check is done once at install time instead of unnecessarily every launch.